mineral-ui/mineral-ui
# [Dropdown] [Menu] Add new components

### Description

Primary updates:
* Add new Dropdown component
* Add new Menu component

Minor updates:
* Add new custom danger, success, and warning icons for internal use
* Updated `mineralTheme` with new variables for `backgroundColor_[variant]_activeMuted`
* Updated Popover to be more composable
* Remove `PureComponent`
* No longer using `react-popper` fork
* Update `react-docgen`

### Motivation and context

New components 🎉 

closes #224 
closes #322 

### Screenshots or videos, if appropriate

Look at the deploy preview:

- https://224-dropdown--mineral-ui.netlify.com/components/dropdown
- https://224-dropdown--mineral-ui.netlify.com/components/menu

### How to test


1. Visit the deploy preview
2. Verify example behavior for Dropdown and Menu components
3. Verify designs with InVision specs

### Types of changes

- New feature (non-breaking change which adds functionality)
- Other (provide details below)

Add new components and refactor some existing

### Checklist
* [x] Automated tests written and passing - **Some Dropdown specs commented out due to enzyme strangeness**
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered - **Plan is to followup with improved a11y support**

### How does this PR make you feel?

![High five](https://media.giphy.com/media/r2BtghAUTmpP2/giphy.gif)
